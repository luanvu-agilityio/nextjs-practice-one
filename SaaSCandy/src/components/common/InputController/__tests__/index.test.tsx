import { useForm } from 'react-hook-form';
import { InputController } from '..';
import { fireEvent, render, screen } from '@testing-library/react';

describe('InputController Component', () => {
  const TestWrapper = ({
    transformValueOnChange,
  }: {
    transformValueOnChange?: (value: string) => string;
  }) => {
    const { control } = useForm({ defaultValues: { testField: '' } });

    return (
      <InputController
        label='Test Input'
        name='testField'
        control={control}
        transformValueOnChange={transformValueOnChange}
      />
    );
  };

  it('matches snapshot', () => {
    const { container } = render(<TestWrapper />);
    expect(container).toMatchSnapshot();
  });

  it('handles input change and transformation', () => {
    const transform = (value: string) => value.toUpperCase();
    render(<TestWrapper transformValueOnChange={transform} />);

    const input = screen.getByLabelText('Test Input');
    fireEvent.change(input, { target: { value: 'hello' } });

    expect(input).toHaveValue('HELLO');
  });
});
