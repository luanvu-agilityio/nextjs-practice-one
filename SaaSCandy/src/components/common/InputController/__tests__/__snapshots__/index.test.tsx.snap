// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`InputController Component > matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-2 w-full"
  >
    <label
      class="text-sm font-medium text-placeholder"
      for="input-«r0»"
    >
      Test Input
    </label>
    <div
      class="flex items-center h-11 w-full min-w-0 rounded-lg border text-md transition-colors bg-transparent px-4 py-2.5 focus-within:border-white border-gray-background"
    >
      <input
        aria-invalid="false"
        class="w-full bg-transparent outline-none placeholder:text-placeholder disabled:pointer-events-none disabled:cursor-not-allowed"
        id="input-«r0»"
        name="testField"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`InputController Component matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-2 w-full"
  >
    <label
      class="text-sm font-medium text-placeholder"
      for="input-«r0»"
    >
      Test Input
    </label>
    <div
      class="flex items-center h-11 w-full min-w-0 rounded-lg border text-md transition-colors bg-transparent px-4 py-2.5 focus-within:border-white border-gray-background"
    >
      <input
        aria-invalid="false"
        class="w-full bg-transparent outline-none placeholder:text-placeholder disabled:pointer-events-none disabled:cursor-not-allowed"
        id="input-«r0»"
        name="testField"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
